### PR TITLE
Local build generate non consumable pom files

### PR DESCRIPTION
**Pull request by talatuyarer**
Current main branch build generate having POM containing ${revision} has version. 

Screenshot 2024-04-30 at 5 26 57 PM

This is neatly explained in the official maven documentation: https://maven.apache.org/maven-ci-friendly.html#install-deploy I updated main pom with official suggestion. Currently it works on my machine.

### CODE REVIEW

**Comment by clmccart** [comment]
should we also add .flattened-pom.xml files to the .gitignore?

**Comment by talatuyarer** [comment]
We sould not. flatten poms are created under target folder which is on .gitignore : https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/main/.gitignore#L1

File: pom.xml [hunk 1]
```diff
@@ -320,6 +320,31 @@ under the License.
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
             </plugin>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>flatten-maven-plugin</artifactId>
+				<version>1.1.0</version>
```
Comment (jayehwhyehentee) [inline pom.xml:326]: Can we use 1.5.0? Its newer and also used by the spark bq connector https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/881d2cca5a6dfe34cc8151a00916bbdcafff21fc/spark-bigquery-parent/pom.xml#L538

File: pom.xml [hunk 2]
```diff
@@ -320,6 +320,31 @@ under the License.
...
+				<configuration>
+					<updatePomFile>true</updatePomFile>
+					<flattenMode>resolveCiFriendliesOnly</flattenMode>
+				</configuration>
+				<executions>
+					<execution>
+						<id>flatten</id>
+						<phase>package</phase>
```
Comment (jayehwhyehentee) [inline pom.xml:334]: Why `package` and not `verify`? Again, referring to the spark connector's setup

**Review by jayehwhyehentee — APPROVED** [review]

**Pull request merged**